pkg/agent/nodeportlocal/portcache/port_table.go

```diff
@@ -80,36 +64,14 @@ func (d *NodePortData) ProtocolInUse(protocol string) bool {
 	return false
 }
 
-func (d *NodePortData) CloseSockets() error {
-	for idx := range d.Protocols {
-		protocolSocketData := &d.Protocols[idx]
-		switch protocolSocketData.State {
-		case stateClosed:
-			// already closed
-			continue
-		case stateInUse:
-			// should not happen
-			return fmt.Errorf("protocol %s is still in use, cannot release socket", protocolSocketData.Protocol)
-		case stateOpen:
-			if err := protocolSocketData.socket.Close(); err != nil {
-				return fmt.Errorf("error when releasing local port %d with protocol %s: %v", d.NodePort, protocolSocketData.Protocol, err)
-			}
-			protocolSocketData.State = stateClosed
-		default:
-			return fmt.Errorf("invalid protocol socket state")
-		}
-	}
-	return nil
-}
-
 type LocalPortOpener interface {
 	OpenLocalPort(port int, protocol string) (io.Closer, error)
 }
 
 type localPortOpener struct{}
 
 type PortTable struct {
-	NodePortTable    map[int]*NodePortData
+	NodePortTable    map[portKey]*NodePortData
```

Comment:
it feels to me like we could have some more unification between Linux & Windows. For example `portKey` could be the same?

Reply:
Updated. Use string type here for both linux and windows.